packages/modules/viewer-react/src/services/iModel/IModelService.ts

```diff
@@ -79,3 +88,80 @@ export const openLocalImodel = async (fileName: string) => {
     return await SnapshotConnection.openFile(fileName);
   }
 };
+
+/**
+ * Generate a viewstate and set it in UiFramework
+ * @param connection \
+ * @param viewportOptions
+ * @param viewCreatorOptions
+ * @param blankConnectionViewState
+ * @returns
+ */
+export const getAndSetViewState = async (
+  connection: IModelConnection,
+  viewportOptions?: ViewerViewportControlOptions,
+  viewCreatorOptions?: ViewerViewCreator3dOptions,
+  blankConnectionViewState?: BlankConnectionViewState
+): Promise<ViewState | undefined> => {
+  const viewState = await getViewState(
+    connection,
+    viewportOptions,
+    viewCreatorOptions,
+    blankConnectionViewState
+  );
+  if (viewState) {
+    UiFramework.setDefaultViewState(viewState);
+  }
+  return viewState;
+};
+
+/**
+ * Generate a viewstate
+ * @param connection
+ * @param viewportOptions
+ * @param viewCreatorOptions
+ * @param blankConnectionViewState
+ * @returns
+ */
+export const getViewState = async (
+  connection: IModelConnection,
+  viewportOptions?: ViewerViewportControlOptions,
+  viewCreatorOptions?: ViewerViewCreator3dOptions,
+  blankConnectionViewState?: BlankConnectionViewState
+): Promise<ViewState | undefined> => {
+  if (!connection.isBlank && connection.isClosed) {
+    return;
+  }
+  let view: ViewState | undefined;
+  if (viewportOptions?.viewState) {
+    if (typeof viewportOptions?.viewState === "function") {
+      view = await viewportOptions?.viewState(connection);
+    } else {
+      view = viewportOptions?.viewState;
+    }
+  }
+  if (
+    !viewportOptions?.alwaysUseSuppliedViewState &&
+    (!view ||
+      (view.iModel.iModelId !== connection.iModelId && connection.isOpen))
+  ) {
+    if (connection.isBlankConnection()) {
```

Comment:
nit: inconsistent with L132,

Reply:
I'll update but they are actually for different cases. The first case we don't care about type guards so the bool suffices. In the second, we are passing a BlankConnection to the function, so isBlankConnection will provide a type guard for us

Reply:
updated